File: backend/api/models/molecule.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class MoleculeBase(BaseModel):
    name: str
    formula: str
    smiles: str  # SMILES notation for molecular structure
    molecular_weight: float

class MoleculeCreate(MoleculeBase):
    pass

# Add this new class
class MoleculeProperties(BaseModel):
    logp: Optional[float] = None
    polar_surface_area: Optional[float] = None
    rotatable_bonds: Optional[int] = None
    h_bond_donors: Optional[int] = None
    h_bond_acceptors: Optional[int] = None

class Molecule(MoleculeBase):
    id: int
    properties: Optional[MoleculeProperties] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    
    class Config:
        from_attributes = True

File: frontend/src/lib/types/molecule.ts
import type { Element } from "./elements";

export interface MoleculeProperties {
    logp?: number;
    polar_surface_area?: number;
    rotatable_bonds?: number;
    h_bond_donors?: number;
    h_bond_acceptors?: number;
}

export interface Molecule {
    id: number;
    
    name: string;
    formula: string;
    smiles: string;
    molecular_weight: number;
    properties?: MoleculeProperties;

    created_at?: string;
    updated_at?: string;
}

export interface MoleculeCreate {
    name: string;
    formula: string;
    smiles: string;
    molecular_weight: number;
}

export interface Formula {
    element: Element;
    count: number;
    custom_mass?: number;
};